cosmos/providers/dbt/parser/project.py

```diff
@@ -213,37 +199,38 @@ class DbtProject:
 
     def __post_init__(self) -> None:
         """
-        Initializes the parser.
+        Parses the dbt models, snapshots, seeds, config yml files and then finally dbt project yml file.
+        For the configuration we go in order of the most specific to the least specific so:
+            1. Config in models.
+            2. Config in the schema.yml files, bottom up in the directory structure.
+            3. Config in the project.yml file defined at the project level.
+            4. If nothing is found for the materialization strategy then view is added for models only to mimic
+                dbt behaviours.
         """
-        # set the project and model dirs
         self.project_dir = Path(os.path.join(self.dbt_root_path, self.project_name))
         self.models_dir = self.project_dir / self.dbt_models_dir
         self.snapshots_dir = self.project_dir / self.dbt_snapshots_dir
         self.seeds_dir = self.project_dir / self.dbt_seeds_dir
-
-        # crawl the models in the project
         for file_name in self.models_dir.rglob("*.sql"):
             self._handle_sql_file(file_name)
-
-        # crawl the snapshots in the project
         for file_name in self.snapshots_dir.rglob("*.sql"):
             self._handle_sql_file(file_name)
-
-        # crawl the seeds in the project
         for file_name in self.seeds_dir.rglob("*.csv"):
             self._handle_csv_file(file_name)
-
-        # crawl the config files in the project
-        for file_name in self.models_dir.rglob("*.yml"):
-            self._handle_config_file(file_name)
+        for root, dirs, files in os.walk(self.models_dir, topdown=False):
+            for file in files:
+                if file.endswith(".yml"):
+                    path = Path(root) / file
+                    self._handle_config_file(path)
```

Comment:
curious why the logic was changed here - does the previous version not work properly?